cpp/assembly.hpp

```diff
@@ -142,4 +211,20 @@ void assemble_matrix(
     throw std::runtime_error("Unsupported integral type");
 };
 
+void assemble_vector(
```

Comment:
In general, I guess we need to start adding doc-strings to our public functions.